### PR TITLE
Proton-C: Add layouts that support 26 key switches with no matrix

### PR DESCRIPTION
**Pull request by Cereal916**
## Description

I added support for multiple switches for the proton-c. This allows people who just want to tinker with custom builds to more easily implement QMK into their board. I also added a couple of layouts for small macro boards / big switch boards.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by e11i0t23** [comment]
Might be worth you looking into https://docs.qmk.fm/#/hardware_avr?id=direct-pin-matrix this allows you to connect each pin to ground and the mcu instead of having a single pin for col

**Comment by Cereal916** [comment]
> Might be worth you looking into https://docs.qmk.fm/#/hardware_avr?id=direct-pin-matrix this allows you to connect each pin to ground and the mcu instead of having a single pin for col

I missed that in the documentation. Thanks, I'll give it a try.

File: keyboards/proton_c/config.h [hunk 1]
```diff
@@ -62,6 +67,11 @@
  */
 //#define FORCE_NKRO
 
+/* key combination for magic key command */
+#define IS_COMMAND() ( \
```
Comment (fauxpark) [inline keyboards/proton_c/config.h]: This block should be removed, as it's the default key combination already defined in QMK.

**Comment by zvecr** [comment]
Just as a heads up, I have just opened https://github.com/qmk/qmk_firmware/pull/6539 after some internal discussions.

To look at the bigger picture, what problem is trying to be solved here? Would better documentation, or more template options to `new_keyboard.sh` help?

**Comment by drashna** [comment]
Personally, I think that a template would be a better option here.  

Otherwise, this feels more like an example than anything else, and should probably go into `/keyboards/handwired/`.

**Comment by zvecr** [comment]
As discussed on Discord, my vote is extended tooling and improved documentation over quite specific examples cluttering up either `keyboards` or `keyboards/handwired`.

**Pull request closed without merging**